snapcraft/internal/project_loader/grammar_processing/_part_grammar_processor.py

```diff
@@ -73,6 +70,16 @@ def __init__(self, *, plugin, properties, project_options, repo):
         self._stage_package_grammar = getattr(plugin, 'stage_packages', [])
         self.__stage_packages = set()
 
+    def get_properties(self):
+        if not self.__properties:
+            self.__properties = self._property_grammar
+            for key, value in self._property_grammar.items():
+                if key in ['source']:
+                    self.__properties[key] = next(iter(grammar.process_grammar(
+                        value if isinstance(value, list) else {value},
+                        self._project_options, True)))
+        return self.__properties
```

Comment:
ok, a copy might not be necessary; but you are not adding the original ones that don't need parsing to self.__properties; is that desired?

Reply:
The original values are assigned initially before the loop. However, you were right suspecting we need to copy, so I added `.copy()` now and a test case to see that `plugin.properties` retains the original properties including grammar (and crucially the test itself needs to `.copy()` so it can verify). Thanks for catching that!

Reply:
Combined with my comment above, I think each property we want grammar-enabled should have its own getter, like stage-packages etc. here. Not the entire set of properties.